vividus-plugin-rest-api/src/main/java/org/vividus/archive/steps/ArchiveSteps.java

```diff
@@ -107,21 +113,37 @@ public void verifyArchiveContainsEntries(DataWrapper archiveData, List<NamedEntr
     {
         Set<String> entryNames = ZipUtils.readZipEntryNamesFromBytes(archiveData.getBytes());
 
+        if (entryNames.isEmpty())
+        {
+        	softAssert.recordFailedAssertion("There are no files in the archive");
+        	return;
+        }
+        attachmentLoaded = false;
         parameters.forEach(entry ->
         {
             String expectedName = entry.getName();
             if (entry.getRule() != null)
             {
                 StringComparisonRule comparisonRule = entry.getRule();
-                softAssert.assertThat(String.format(
-                        "The archive contains entry matching the comparison rule '%s' with name pattern '%s'",
-                        comparisonRule, expectedName), entryNames, hasItem(comparisonRule.createMatcher(expectedName)));
+				if (!softAssert.assertThat(String.format(
```

Comment:
why not to just assign result of softAssert.assertThat(....) to a Boolean variable?

Reply:
because I'll lose the result if last entry matching the comparison rule